keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -190,21 +190,25 @@ class SqlArtifactRepository(
     } == 1
   }
 
-  override fun updateArtifactMetadata(name: String, type: ArtifactType, version: String, status: ArtifactStatus?, artifactMetadata: ArtifactMetadata) {
+  override fun updateArtifactMetadata(name: String, type: ArtifactType, version: String, status: ArtifactStatus?, artifactMetadata: ArtifactMetadata) : Boolean {
     if (!isRegistered(name, type)) {
       throw NoSuchArtifactException(name, type)
     }
 
-    return sqlRetry.withRetry(WRITE) {
+    val changed =  sqlRetry.withRetry(WRITE) {
       jooq.update(ARTIFACT_VERSIONS)
         .set(ARTIFACT_VERSIONS.BUILD_METADATA, objectMapper.writeValueAsString(artifactMetadata.buildMetadata))
         .set(ARTIFACT_VERSIONS.GIT_METADATA, objectMapper.writeValueAsString(artifactMetadata.gitMetadata))
-        .where(ARTIFACT_VERSIONS.NAME.eq(name))
-        .and(ARTIFACT_VERSIONS.TYPE.eq(type))
-        .and(ARTIFACT_VERSIONS.VERSION.eq(version))
+
+        .where(
+          ARTIFACT_VERSIONS.NAME.eq(name),
+          ARTIFACT_VERSIONS.TYPE.eq(type),
+          ARTIFACT_VERSIONS.VERSION.contains(version))
```

Comment:
this is the major change ^. let me know if there's a better way for doing this...

Reply:
AFAIK, this is only a problem for Debians for historical reasons. The way we've approached handling it elsewhere is slightly different, for example:
https://github.com/spinnaker/keel/blob/9c83039ce03194841c27d6665cfb1304d3e77b60/keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt#L71-L78

So, another way of doing this would be something like:
```suggestion
          ARTIFACT_VERSIONS.VERSION.eq(version).or(ARTIFACT_VERSIONS.VERSION.eq("$name-$version"))
```